### ISSUE TITLE
Failure to Compile on Gentoo

### ISSUE THREAD

**Issue by sdfg2** [ONLYOFFICE/DocumentServer#120]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
Compile error

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**
Follow the guide until core/make

**What is the expected behavior?**
It compiles

**Which versions of DocumentServer, and which browser / OS are affected by this issue? Did this work in previous versions of DocumentServer?**
commit 9a2cf4ff921fcefb52bac0a24837ea05ab948fdc.  Not installed previously.

Error:
`cd /root/DocumentServer/core/UnicodeConverter/UnicodeConverter.build/ && make -j 2;
make[1]: Entering directory '/root/DocumentServer/core/UnicodeConverter/UnicodeConverter.build'
g++ -c -pipe -fvisibility=hidden -O2 -Wall -W -D_REENTRANT -fPIC -DLINUX -D_LINUX -DUNICODECONVERTER_USE_DYNAMIC_LIBRARY -DQT_NO_DEBUG -DQT_PLUGIN -I/usr/share/qt4/mkspecs/linux-g++ -I../../UnicodeConverter -I/usr/include/qt4 -I/root/DocumentServer/core/Common/3dParty/icu/linux_64/usr/local/include -I../core_build/linux_64/release/moc -I. -o ../core_build/linux_64/release/obj/UnicodeConverter.o ../UnicodeConverter.cpp
In file included from ../UnicodeConverter.cpp:37:0:
../../DesktopEditor/common/File.h: In member function ‘bool NSFile::CFileBinary::WriteFile(const BYTE*, DWORD)’:
../../DesktopEditor/common/File.h:711:11: warning: unused variable ‘nCountWrite’ [-Wunused-variable]
    size_t nCountWrite = fwrite((const void*)pData, 1, nBytesCount, m_pFile);
           ^
../../DesktopEditor/common/File.h: In static member function ‘static std::wstring NSFile::CFileBinary::GetTempPath()’:
../../DesktopEditor/common/File.h:972:32: error: ‘getenv’ was not declared in this scope
    char *folder = getenv("TEMP");
                                ^
../../DesktopEditor/common/File.h:980:12: warning: deprecated conversion from string constant to ‘char*’ [-Wwrite-strings]
     folder = "/tmp";
            ^
../../DesktopEditor/common/File.h: In static member function ‘static std::wstring NSFile::CFileBinary::CreateTempFileWithUniqueName(const wstring&, const wstring&)’:
../../DesktopEditor/common/File.h:1001:27: error: ‘mkstemp’ was not declared in this scope
    int res = mkstemp(pcRes);
                           ^
../../DesktopEditor/common/File.h: In static member function ‘static bool NSFile::CFileBinary::OpenTempFile(std::wstring*, FILE**, wchar_t*, wchar_t*, wchar_t*, wchar_t*)’:
../../DesktopEditor/common/File.h:1023:35: error: ‘getenv’ was not declared in this scope
    if ((wsTempDirA = getenv("TEMP")) && (wsFolder == NULL))
                                   ^
../../DesktopEditor/common/File.h:1044:23: error: ‘to_wstring’ is not a member of ‘std’
     wsFileName.append(std::to_wstring(nTime + nIndex));
                       ^
In file included from /usr/include/string.h:32:0,
                 from ../../DesktopEditor/common/File.h:52,
                 from ../UnicodeConverter.cpp:37:
../../DesktopEditor/common/File.h: At global scope:
../../DesktopEditor/common/File.h:1009:136: warning: unused parameter ‘wsName’ [-Wunused-parameter]
   static bool OpenTempFile(std::wstring *pwsName, FILE **ppFile, wchar_t *wsMode, wchar_t *wsExt, wchar_t *wsFolder, wchar_t* wsName = NULL)
                                                                                                                                        ^
../UnicodeConverter.cpp: In member function ‘std::string NSUnicodeConverter::CUnicodeConverter_Private::fromUnicode(const wchar_t*, const unsigned int&, const char*)’:
../UnicodeConverter.cpp:90:78: error: ‘malloc’ was not declared in this scope
                 UChar* pUChar = (UChar*)malloc(nUCharCapacity * sizeof(UChar));
                                                                              ^
../UnicodeConverter.cpp: In member function ‘std::wstring NSUnicodeConverter::CUnicodeConverter_Private::toUnicode(const char*, const unsigned int&, const char*)’:
../UnicodeConverter.cpp:137:81: error: ‘malloc’ was not declared in this scope
                     UChar* targetStart = (UChar*)malloc(uBufSize * sizeof(UChar));
                                                                                 ^
../UnicodeConverter.cpp:128:21: warning: unused variable ‘iii’ [-Wunused-variable]
                 int iii = ucnv_getCCSID(conv, &status);
                     ^
In file included from ../UnicodeConverter.cpp:37:0:
../../DesktopEditor/common/File.h: At global scope:
../../DesktopEditor/common/File.h:1170:25: warning: ‘std::wstring NSFile::GetProcessDirectory()’ defined but not used [-Wunused-function]
     static std::wstring GetProcessDirectory()
                         ^
../../DesktopEditor/common/File.h:1197:25: warning: ‘std::wstring NSFile::GetFileExtention(const wstring&)’ defined but not used [-Wunused-function]
     static std::wstring GetFileExtention(const std::wstring& sPath)
                         ^
../../DesktopEditor/common/File.h:1204:25: warning: ‘std::wstring NSFile::GetFileName(const wstring&)’ defined but not used [-Wunused-function]
     static std::wstring GetFileName(const std::wstring& sPath)
                         ^
../../DesktopEditor/common/File.h:1223:25: warning: ‘std::wstring NSFile::GetDirectoryName(const wstring&)’ defined but not used [-Wunused-function]
     static std::wstring GetDirectoryName(const std::wstring& sPath)
                         ^
make[1]: *** [Makefile:218: ../core_build/linux_64/release/obj/UnicodeConverter.o] Error 1
make[1]: Leaving directory '/root/DocumentServer/core/UnicodeConverter/UnicodeConverter.build'
make: *** [Makefile:241: build/lib/linux_/libUnicodeConverter.so*] Error 2
`

gcc -v 

`Using built-in specs.
COLLECT_GCC=/usr/x86_64-pc-linux-gnu/gcc-bin/4.9.4/gcc
COLLECT_LTO_WRAPPER=/usr/libexec/gcc/x86_64-pc-linux-gnu/4.9.4/lto-wrapper
Target: x86_64-pc-linux-gnu
Configured with: /var/tmp/portage/sys-devel/gcc-4.9.4/work/gcc-4.9.4/configure --host=x86_64-pc-linux-gnu --build=x86_64-pc-linux-gnu --prefix=/usr --bindir=/usr/x86_64-pc-linux-gnu/gcc-bin/4.9.4 --includedir=/usr/lib/gcc/x86_64-pc-linux-gnu/4.9.4/include --datadir=/usr/share/gcc-data/x86_64-pc-linux-gnu/4.9.4 --mandir=/usr/share/gcc-data/x86_64-pc-linux-gnu/4.9.4/man --infodir=/usr/share/gcc-data/x86_64-pc-linux-gnu/4.9.4/info --with-gxx-include-dir=/usr/lib/gcc/x86_64-pc-linux-gnu/4.9.4/include/g++-v4 --with-python-dir=/share/gcc-data/x86_64-pc-linux-gnu/4.9.4/python --enable-languages=c,c++,fortran --enable-obsolete --enable-secureplt --disable-werror --with-system-zlib --enable-nls --without-included-gettext --enable-checking=release --with-bugurl=https://bugs.gentoo.org/ --with-pkgversion='Gentoo 4.9.4 p1.0, pie-0.6.4' --enable-libstdcxx-time --enable-shared --enable-threads=posix --enable-__cxa_atexit --enable-clocale=gnu --enable-multilib --with-multilib-list=m32,m64 --disable-altivec --disable-fixed-point --enable-targets=all --disable-libgcj --enable-libgomp --disable-libmudflap --disable-libssp --disable-libcilkrts --enable-vtable-verify --enable-libvtv --enable-lto --without-cloog --enable-libsanitizer
Thread model: posix
gcc version 4.9.4 (Gentoo 4.9.4 p1.0, pie-0.6.4) 
`

**Comment by ShockwaveNN** [comment]
@sdfg2 Yeah, I was able to reproduce your problem. We try to figure something out

**Comment by ShockwaveNN** [comment]
Tryed to reproduce it again, now there is trobules with compilation of v8 under gcc-6.x, see WIP Docker at https://github.com/onlyoffice-testing-robot/documentserver-from-source-build/pull/9
